src/librustc_errors/emitter.rs

```diff
@@ -703,6 +703,40 @@ impl EmitterWriter {
         }
     }
 
+    /// Add a left margin to every line but the first, given a padding length and the label being
+    /// displayed.
+    fn msg_with_padding(&self, msg: &str, padding: usize, label: &str) -> String {
```

Comment:
It seems like the labels attached to a span might also need wrapping, not just notes, right?

But I guess this code is not about "autowrap" but specifically targeting `\n` that are embedded in a message. I see, carry on.